giskard/project.py

```diff
@@ -378,3 +379,11 @@ def validate_df(df: pd.DataFrame, column_types) -> pd.DataFrame:
                 ):
                     df[column] = df[column].astype(float)
             return df
+
+    @staticmethod
+    def _verify_category_columns(df: pd.DataFrame, column_types):
+        for name, types in column_types.items():
+            if types == SupportedColumnType.CATEGORY.value and len(df[name].unique()) > 30:
+                warnings.warn(f"{name} column has more than 30 categories. "
+                              f"Are you sure it’s a “category” feature and not a “text” feature ? If No, turn "
+                              f"this feature type into “text” in column_types")
```

Comment:
why only "text"? This can happen when a numerical variable is wrongly marked as categorical.

Maybe we can say "Categorical feature X contains N distinct values. If necessary use 'numeric' or 'text' types instead".

Reply:
Totally agree with the comment . I ll update it